### PR TITLE
SSA tests

### PR DESCRIPTION
**Pull request by higs4281**
This sets up an automated test of returned SSA values, for the purpose of detecting changes in SSA benefit formulas.  
The changes shouldn't break the app, because we accept whatever SSA calculates, but it will give our subject matter experts a warning when formulas change.

## Additions

- utility `ssa_check.py`
- Calibration model to store expected SSA values
- tests to match

## Testing

For local testing, run migrations to create the new Calibration table and load the new Calibration object
  ```bash
  ./manage.py schemamigration retirement_api --auto
  ./manage.py migrate retirement_api
  ./manage.py loaddata retiredata.json
  ```

To test the script, you can fire up the Django shell in standalone and run the script. It should run through 19 SSA tests, including known edge cases, and check all variables and bar-chart values. 
  ```bash
  ./manage.py shell
  ```
  ```python
  from retirement_api.utils.ssa_check import *
  run_tests()
  ```
**Note**: You may see some mismatch errors for the `born-on-1st-age-46` test. This is a known false mismatch.

## Review

- @amymok -- You don't need to install the app and run the tests, unless you really want to. 
- @marteki @niqjohnson @mistergone -- would be great to have at least one run the script locally. 

## Todos

- create two management commands to run the script and to recalibrate it. 

## Checklist

* [x] CHANGELOG has been updated

### CODE REVIEW

File: retirement_api/fixtures/retiredata.json [hunk 1]
```diff
@@ -1,6 +1,14 @@
...
...r\": 2037, \"yob\": 1970, \"lastEarn\": \"\", \"dobday\": 3, \"retiremonth\": 4, \"earnings\": 40000}, \"months_past_birthday\": 0, \"full retirement age\": \"67\", \"survivor benefits\": {\"spouse at full retirement age\": \"\", \"family maximum\": \"\", \"spouse caring for child\": \"\", \"child\": \"\"}}}, \"born_on_3rd_in_1948\": {\"note\": \"Age 68 is past your full benefit claiming age.\", \"params_adjusted\": false, \"current_age\": 68, \"past_fra\": true, \"error\": \"\", \"data\": {\"benefits\": {\"age 63\": 0, \"age 62\": 0, \"age 67\": 0, \"age 66\": 0, \"age 65\": 0, \"age 64\": 0, \"age 69\": 1463.0, \"age 68\": 1355, \"age 70\": 1571.0}, \"disability\": \"You have reached full retirement age and are not eligible for disability benefits.\", \"early retirement age\": \"\", \"params\": {\"prgf\": 2, \"lastYearEarn\": \"\", \"dobmon\": 4, \"dollars\": 1, \"retireyear\": 2016, \"yob\": 1948, \"lastEarn\": \"\", \"dobday\": 3, \"retiremonth\": 4, \"earnings\": 40000}, \"months_past_birthday\": 0, \"full retirement age\": \"66\", \"survivor benefits\": {\"spouse at full retirement age\": \"\", \"family maximum\": \"\", \"spouse caring for child\": \"\", \"child\": \"\"}}}}",
```
Comment (amymok) [inline retirement_api/fixtures/retiredata.json]: May want to break this line up for readable (not highest priority though).
http://stackoverflow.com/questions/2058925/how-can-i-break-up-this-long-line-in-python

File: retirement_api/fixtures/test_calibration.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
...r\": 2037, \"yob\": 1970, \"lastEarn\": \"\", \"dobday\": 3, \"retiremonth\": 4, \"earnings\": 40000}, \"months_past_birthday\": 0, \"full retirement age\": \"67\", \"survivor benefits\": {\"spouse at full retirement age\": \"\", \"family maximum\": \"\", \"spouse caring for child\": \"\", \"child\": \"\"}}}, \"born_on_3rd_in_1948\": {\"note\": \"Age 68 is past your full benefit claiming age.\", \"params_adjusted\": false, \"current_age\": 68, \"past_fra\": true, \"error\": \"\", \"data\": {\"benefits\": {\"age 63\": 0, \"age 62\": 0, \"age 67\": 0, \"age 66\": 0, \"age 65\": 0, \"age 64\": 0, \"age 69\": 1463.0, \"age 68\": 1355, \"age 70\": 1571.0}, \"disability\": \"You have reached full retirement age and are not eligible for disability benefits.\", \"early retirement age\": \"\", \"params\": {\"prgf\": 2, \"lastYearEarn\": \"\", \"dobmon\": 4, \"dollars\": 1, \"retireyear\": 2016, \"yob\": 1948, \"lastEarn\": \"\", \"dobday\": 3, \"retiremonth\": 4, \"earnings\": 40000}, \"months_past_birthday\": 0, \"full retirement age\": \"66\", \"survivor benefits\": {\"spouse at full retirement age\": \"\", \"family maximum\": \"\", \"spouse caring for child\": \"\", \"child\": \"\"}}}}",
```
Comment (amymok) [inline retirement_api/fixtures/test_calibration.json]: Same for this.

File: retirement_api/tests/test_models.py [hunk 3]
```diff
@@ -38,9 +49,10 @@ def test_question_translist(self):
     def test_quesiton_dump(self):
```
Comment (amymok) [inline retirement_api/tests/test_models.py]: misspelled question

File: retirement_api/utils/ssa_check.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    'dobmon': 0,
+    'dobday': 0,
+    'yob': 0,
+    'earnings': 40000,
+    'lastYearEarn': '',
+    'lastEarn': '',
+    'retiremonth': '',
+    'retireyear': '',
+    'dollars': 1,
+    'prgf': 2
```
Comment (amymok) [inline retirement_api/utils/ssa_check.py]: What does prgf stand for?

Comment (higs4281) [inline retirement_api/utils/ssa_check.py]: a good question. the param names are defined by SSA, so we can't change them. i couldn't find documentation on what that param specifies. I can see that all posts to the Quick Calculator have that value. I'll ask our contacts.

File: retirement_api/fixtures/test_calibration.json (hunk 2 above)
Comment (higs4281) [inline retirement_api/fixtures/test_calibration.json]: we use indent=4 for our main fixture, but when I used indenting on the calibration file, it played  havoc with tests. still pursuing. 
over time calibration files will multiply, and I don't foresee us storing them in this repo. This is just a starter.
